tycho-plugins/repository-utils/src/main/java/org/jboss/tools/tycho/sitegenerator/CompareWithBaselineMojo.java

```diff
@@ -0,0 +1,128 @@
+
+import org.apache.commons.io.FileUtils;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.component.annotations.Requirement;
+import org.codehaus.plexus.logging.Logger;
+import org.eclipse.sisu.equinox.EquinoxServiceFactory;
+import org.eclipse.tycho.ReactorProject;
+import org.eclipse.tycho.artifacts.TargetPlatform;
+import org.eclipse.tycho.core.ee.shared.ExecutionEnvironmentConfigurationStub;
+import org.eclipse.tycho.core.osgitools.DefaultReactorProject;
+import org.eclipse.tycho.core.shared.TargetEnvironment;
+import org.eclipse.tycho.osgi.adapters.MavenLoggerAdapter;
+import org.eclipse.tycho.p2.resolver.facade.P2ResolutionResult;
+import org.eclipse.tycho.p2.resolver.facade.P2ResolutionResult.Entry;
+import org.eclipse.tycho.p2.resolver.facade.P2Resolver;
+import org.eclipse.tycho.p2.resolver.facade.P2ResolverFactory;
+import org.eclipse.tycho.p2.target.facade.TargetPlatformConfigurationStub;
+import org.osgi.framework.Version;
+
+@Mojo(defaultPhase=LifecyclePhase.VERIFY, requiresProject=false, name="compare-version-with-baselines")
+public class CompareWithBaselineMojo extends AbstractMojo {
+
+	@Parameter(property="project", readonly=true)
+	private MavenProject project;
+
+	@Parameter(property="baselines", name="baselines")
+	private List<String> baselines;
+
+	@Parameter(property="skip")
+	private boolean skip;
+
+	@Requirement
+	@Component
+	private EquinoxServiceFactory equinox;
+
+	@Component
+	private Logger plexusLogger;
+
+	public void execute() throws MojoExecutionException, MojoFailureException {
+		if (skip) {
+			getLog().info("Skipped");
+			return;
+		}
+		ReactorProject reactorProject = DefaultReactorProject.adapt(project);
+		Set<?> dependencyMetadata = reactorProject.getDependencyMetadata(true);
+		if (dependencyMetadata == null || dependencyMetadata.isEmpty()) {
+			getLog().info("No p2 artifacts created");
+			return;
+		}
+
+		P2ResolverFactory resolverFactory = this.equinox.getService(P2ResolverFactory.class);
+		P2Resolver resolver = resolverFactory.createResolver(new MavenLoggerAdapter(this.plexusLogger, true));
+
+		TargetPlatformConfigurationStub baselineTPStub = new TargetPlatformConfigurationStub();
+		baselineTPStub.setEnvironments(Collections.singletonList(TargetEnvironment.getRunningEnvironment()));
+		for (String baselineRepo : this.baselines) {
+			baselineTPStub.addP2Repository(toRepoURI(baselineRepo));
+		}
+		TargetPlatform baselineTP = resolverFactory.getTargetPlatformFactory().createTargetPlatform(baselineTPStub, new ExecutionEnvironmentConfigurationStub("JavaSE-1.8"), null, null);
```

Comment:
why hardcoded java 1.8?  should it not use the environment specified in the build ?